### DISCUSSION TITLE
Guided tour

### DISCUSSION THREAD

**Discussion by meditadvisors** [directus/directus#16982] [Feature Requests]
Happy new year!

I would like to provide a guided tour of the app, eg. for first time users or on demand, and I tried using [https://shepherdjs.dev/](https://shepherdjs.dev/)

I use a Directus docker instance and an extension with custom JS embeds, see code below 

The PoC kind of works but I am not able to access the CSS selectors, e.g.  element: '#sidebar'

1. Any ideas how I can access various CSS selectors in the hook?
2. How can I trigger the tour only after the user logs in?
3. How can I add a button or something similar in the user menu or sidebar to trigger the tour?

Any (other) feedback on how to do an app tour (with or without shepherdjs) is welcomed! (maybe this could be a core feature?)

Thanks!

PS: the docker instance requires a custom CSP: 
`- CONTENT_SECURITY_POLICY_DIRECTIVES__SCRIPT_SRC="'self' 'unsafe-eval' 'unsafe-inline' https://cdnjs.cloudflare.com/ajax/libs/shepherd.js/11.0.1/js/"`
Importing the module or [vue-sheperd](https://github.com/shipshapecode/vue-shepherd) did not work for me, but may be related to my poor coding skills

```
export default ({ embed }, { env }) => {
	// Tour of the app
	embed(
		'head',
		'<script src="https://cdnjs.cloudflare.com/ajax/libs/shepherd.js/11.0.1/js/shepherd.min.js" integrity="sha512-TpfSb4OAUaETX8uPvA1AMsPmJQK+deMNkK0+EZ/vZg91KHSm9dxfynnC0StGzFADMJftzCWll0/7USvjy83NRQ==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/shepherd.js/11.0.1/css/shepherd.css" integrity="sha512-Nt4iMiqRumUbivLZoP7YOWEI5unLFL7vgncYCX/PnzAyXSceWf1o6sPIgwuIiWfMyXSHLQt0+UwD2nKBheiSVQ==" crossorigin="anonymous" referrerpolicy="no-referrer" />'
	);

	embed(
		'body',
		() => `<script>
		const tour = new Shepherd.Tour({
			useModalOverlay: true,
			defaultStepOptions: {
			  classes: 'shadow-md bg-purple-dark',
			  scrollTo: true
			}
		  });

		  tour.addStep({
			id: 'tour-step-1',
			text: 'Welcome! Click next for the app tour',
			attachTo: {
				element: '#sidebar',
				on: 'left'
			  },
			buttons: [
			  {
				text: 'Next',
				action: tour.next
			  }
			]
		  });

		  tour.addStep({
			id: 'tour-step-2',
			text: 'First, a bit of intro...',
			buttons: [
			  {
				text: 'Next',
				action: tour.next
			  }
			]
		  });

		  tour.start();

	  </script>`
	);
};
```

**Comment by br41nslug** [comment]
> Any ideas how I can access various CSS selectors in the hook?

This is not possible as the hook runs on the server and the CSS selectors are in the browser, you'd have to `embed` the logic using selectors like the tour itself.

Im not sure if the other 2 questions are possible from the outside.

**Reply by meditadvisors** [reply]
thanks for confirmation

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
